base/compiler/optimize.jl

```diff
@@ -631,6 +638,9 @@ function slot2reg(ir::IRCode, ci::CodeInfo, sv::OptimizationState)
     defuse_insts = scan_slot_def_use(nargs, ci, ir.stmts.inst)
     𝕃ₒ = optimizer_lattice(sv.inlining.interp)
     @timeit "construct_ssa" ir = construct_ssa!(ci, ir, domtree, defuse_insts, sv.slottypes, 𝕃ₒ) # consumes `ir`
+    # NOTE now we have converted `ir` to the SSA form and eliminated slots
+    # let's resize `argtypes` now and remove unnecessary types for the eliminated slots
+    resize!(ir.argtypes, nargs)
```

Comment:
I believe this was not done before because it was thought to harm the usefulness of inspection tooling (like code_warntype) while not providing any benefit to the compiler (which presumably can drop this entirely)

Reply:
Once we're at the IR level, the slots are gone anyway, I don't see what this information could be good for.

Reply:
Yes. `code_warntype` works with this PR since it doesn't hit the optimization pipeline. If extra tooling like Cthulhu wants to keep unoptimized `CodeInfo` for later inspection, it should make additional efforts to preserve its state before optimization, which otherwise may be modified by the optimization pipeline.